core/workflow/loadDPL.go

```diff
@@ -0,0 +1,58 @@
+/*
+ * === This file is part of ALICE O² ===
+ *
+ * Copyright 2020 CERN and copyright holders of ALICE O².
+ * Author: Ayaan Zaidi <azaidi@cern.ch>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * In applying this license CERN does not waive the privileges and
+ * immunities granted to it by virtue of its status as an
+ * Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+package workflow
+
+import (
+	"github.com/AliceO2Group/Control/common/gera"
+	"github.com/AliceO2Group/Control/core/task"
+)
+
+func LoadDPL(tasks []*task.Class) (workflow Role, err error) {
+	root := new(aggregatorRole)
+
+	for _, taskItem := range tasks {
+		SingleTaskRole := taskRole{
+			roleBase: roleBase{
+				Name:        taskItem.Identifier.Name,
+				Connect:     nil,
+				Constraints: nil,
+				Defaults:    gera.MakeStringMap(),
+				Vars:        gera.MakeStringMap(),
+				Bind:        nil,
+			},
+		}
+
+		SingleTaskRole.Connect = taskItem.Connect
```

Comment:
As before, please take care not to do this.
`Connect` here is a slice, which means it behaves like a pointer type. You are not copying the `Connect` entries, you are just passing the pointer. This is not what we want. `Connect` entries should be generated, and the `task.Class`'s `Connect` slice can be used as base for this, but if so, it needs to be a deep copy and not a pointer copy.

You can achieve this with `append`.

Reply:
My bad, I misunderstood. I thought assigning during declaration of the `SingleTaskRole` meant that the pointer would be passed. I expected assigning it outside the declaration to be different. I'll have this fixed.